sdks/typescript/src/apache_beam/coders/standard_coders.ts

```diff
@@ -123,9 +123,9 @@ export class DoubleCoder implements Coder<number> {
   }
 
   decode(reader: Reader, context: Context): number {
-    const barr = new Uint8Array(reader.buf, reader.pos, 8);
-    const dView = new DataView(barr.buffer);
-    reader.float();
+    const barr = new Uint8Array(reader.buf);
+    const dView = new DataView(barr.buffer.slice(reader.pos, reader.pos + 8));
```

Comment:
this seems to add a new allocation plus copy (via `slice`) to the decode function, right? what's the issue that this solves? Is there a way to solve this without the extra copy?

Reply:
The problem was that despite constructing the `barr` as new `Uint8Array(reader.buf, reader.pos, 8)`, `barr.buffer` still refers to the entire (non-cropped) underlying buffer. This means the coder failed whenever the value was not the first element in the stream (as it is in js_coders, hence the added test, as well as in iterables, as a second element, etc.) I didn't see a way to avoid the extra copy, but it's only 8 bytes (as opposed to an unbounded amount) so hopefully that should be relatively cheap.